### PR TITLE
Vscode extension packaging

### PR DESCRIPTION
**Pull request by Goju-Ryu**
This PR sets up the vs-code extension to make it easy to package.

I also added and changed a few things:
- Replaced template text in README with text relevant the numbat extension
- Added extension Icon
- Used numbat icon for numbat files
- Added license, repo,  and homepage information
- Added keywords
- Redesigned how string highlighting is passed (this fixes some highlighting bugs and adds support for  multi line strings) 

**What it looks like:**
image
image

### CODE REVIEW

**Comment by Goju-Ryu** [comment]
Installing it is very easy and can be summed up in a single picture: 
image
Click `Install from VSIX...` select the VSIX file in the file selector that pops up and it is automatically installed.

That said, it is currently not easy to get the VSIX file without getting the repo and creating it for yourself. I am not sure what the best way to solve it is, but an easy solution is to release it with the other numbat assets in each release. I will update the README and Numbat documentation to reflect how we distribute the VSIX when we have decided.

File: vscode-extension/README.md [hunk 1]
```diff
@@ -1,65 +1,28 @@
...
-
-For example if there is an image subfolder under your extension project workspace:
-
-\!\[feature X\]\(images/feature-x.png\)
-
-> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.
+- Syntax highlighting
+- Code block folding
+- comment toggling
+- autosurounding brackets
```
Comment (sharkdp) [inline vscode-extension/README.md:10]: Minor: 
```suggestion
- autosurrounding brackets
```

**Review by sharkdp — APPROVED** [review]
This is great! Thank you very much.

**Comment by sharkdp** [comment]
> an easy solution is to release it with the other numbat assets in each release.

:+1: 

> I will update the README and Numbat documentation to reflect how we distribute the VSIX when we have decided.

That would be fantastic

**Pull request merged**